### PR TITLE
KAFKA-16406: Splitting consumer integration test

### PR DESCRIPTION
**Pull request by lianetm**
Splitting consumer integration tests to allow for parallelization and reduce build times. This PR is only extracting tests from PlainTextConsumerTest into separate files, no changes in logic. Grouping tests by the feature they relate to so that they can be easily found

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/PlaintextConsumerAssignTest.scala
Comment (lianetm) [inline core/src/test/scala/integration/kafka/api/PlaintextConsumerAssignTest.scala:246]: FYI, this section (here and in all other added filed) is the only addition introduced by this PR just as a way to have access to the functions in BaseConsumerTest without having to inherit (like the the PlaintextConsumer does). Avoiding extending the BaseConsumerTest to make sure we don't end up running the tests defined in BaseConsumerTest repeatedly.

**Comment by lianetm** [comment]
Hey @lucasbru , could you take a look if you have a chance? No changes in logic other than the addition I pointed out (see comment above). With this split I've been getting the run times for the main PlainText under 1h (used to between 3h-4h), and we're adding several new integration tests, all under 30mins. We'll get more data as we continue to run builds with the change but should be a good step moving forward and looking for parallelization. Thanks!

**Review by AndrewJSchofield — APPROVED** [review]
I think this is very sensible. I wonder whether it would actually be a good idea to remove `PlaintextConsumerTest` entirely and move every single test case into one of the categorised tests. Just an idle thought, and I'll happily approve like this.

File: core/src/test/scala/integration/kafka/api/PlaintextConsumerAssignTest.scala
Comment (lucasbru) [reply core/src/test/scala/integration/kafka/api/PlaintextConsumerAssignTest.scala:246]: Thanks for this. Too often, we are using inheritance to get one or two utility methods

**Comment by lucasbru** [comment]
@AndrewJSchofield I think Lianet did not remove the `PlaintextConsumerTest` to makes sure we still run the tests in `BaseConsumerTest` in the plaintext case once. I guess we could make one of the categorized test suites inherit from `BaseConsumerTest`, but I think I'd prefer more just keeping `PlaintextConsumerTest`.

**Review by lucasbru — APPROVED** [review]
LGTM, thanks! Hard to tell what is changed from the diff. I'm assuming you did an exact copy, and I checked that we are not losing any test cases in this change. @lianetm I'll let you take a look at andrews comment before merging this.

**Comment by lianetm** [comment]
Thanks both for the comments. I do like @AndrewJSchofield 's idea of having it all categorized but I ended up with several unrelated tests that seemed hard to group in a sensible way. Also the fact that @lucasbru pointed out, that I wanted to keep the way of running the abstract `BaseConsumerTest` made me keep the `PlaintextConsumerTest` as a kind of generic one. Makes sense? 

@lucasbru just to confirm, I did an exact copy of the contents, only changes were the additions mentioned in the comments. 

For the record, I think there's more room for improvement with a similar approach in other tests, I will follow up on that, probably following with the `PlaintextAdminIntegrationTest` which is in a similar (even worse) situation regarding run times.

Thanks!

**Comment by lucasbru** [comment]
Two related flaky tests, `kafka.api.PlaintextConsumerTest.testExpandingTopicSubscriptions(String, String)[4]` and `org.apache.kafka.tools.consumer.group.DescribeConsumerGroupTest.testDescribeExistingGroupWithNoMembers(String, String)[4]`. But doesn't seem to be caused by this PR, as they have flaked on trunk before.

**Pull request merged**
